community/server/src/main/java/org/neo4j/server/configuration/ConfigLoader.java

```diff
@@ -55,10 +55,17 @@ public ConfigLoader( List<Class<?>> settingsClasses )
 
     public Config loadConfig( Optional<File> configFile, Log log, Pair<String, String>... configOverrides )
     {
-        return loadConfig( configFile, log, settings -> {}, configOverrides );
+        return loadConfig( Optional.empty(), configFile, log, settings -> {}, configOverrides );
     }
 
-    public Config loadConfig( Optional<File> configFile,
+    public Config loadConfig( File homeDir, Optional<File> configFile, Log log,
+            Pair<String, String>... configOverrides )
+    {
+        return loadConfig( Optional.ofNullable( homeDir ), configFile, log, settings -> {}, configOverrides );
+    }
+
+    public Config loadConfig( Optional<File> homeDir,
+                              Optional<File> configFile,
                               Log log,
                               Consumer<Map<String, String>> customizer,
                               Pair<String, String>... configOverrides )
```

Comment:
Notably, ServerBootstrapper and DesktopConfigurator. And a dozen times in tests.